diff --git a/mk/elftoolchain.mk b/mk/elftoolchain.mk
index 55cbe540..8fd64794 100644
--- a/mk/elftoolchain.mk
+++ b/mk/elftoolchain.mk
@@ -34,8 +34,6 @@ export CFLAGS   := -O2 -g -fdata-sections -ffunction-sections -fstack-protector
 
 $(SRCS_DIR)/.download_stamp:
 	$(MKDIR) -p $(SRCS_DIR)
-	$(CURL) --progress-bar -fSL $(URL) | \
-	$(TAR) -C $(SRCS_DIR) --strip-components=1 -xj $(addprefix $(PREFIX)/,mk common libelf)
 	$(CP) $(MAKE_DIR)/native-elf-format $(COMMON)
 	@touch $@
 
diff --git a/mk/nvidia-modprobe.mk b/mk/nvidia-modprobe.mk
index 34062224..03aa5277 100644
--- a/mk/nvidia-modprobe.mk
+++ b/mk/nvidia-modprobe.mk
@@ -33,8 +33,6 @@ LIB_OBJS := $(LIB_SRCS:.c=.o)
 
 $(SRCS_DIR)/.download_stamp:
 	$(MKDIR) -p $(SRCS_DIR)
-	$(CURL) --progress-bar -fSL $(URL) | \
-	$(TAR) -C $(SRCS_DIR) --strip-components=1 -xz $(PREFIX)/modprobe-utils
 	$(PATCH) -d $(SRCS_DIR) -p1 < $(PATCH_FILE)
 	@touch $@
 
